crates/bevy-inspector-egui/src/bevy_inspector/mod.rs

```diff
@@ -605,66 +606,119 @@ pub(crate) fn ui_for_entity_components(
+                let mut env = InspectorUi::for_bevy(type_registry, &mut cx);
+                let id = id.with(component_id);
+                let options = &();
+
+                match value {
+                    ReflectBorrow::Mutable(mut value) => {
+                        let changed = env.ui_for_reflect_with_options(
+                            value.bypass_change_detection().as_partial_reflect_mut(),
+                            ui,
+                            id,
+                            options,
+                        );
+
+                        if changed {
+                            value.set_changed();
+                        }
                     }
-                }
-                ReflectBorrow::Immutable(value) => env.ui_for_reflect_readonly_with_options(
-                    value.as_partial_reflect(),
-                    ui,
-                    id,
-                    options,
-                ),
-            };
-        });
+                    ReflectBorrow::Immutable(value) => match value.reflect_clone() {
+                        Err(_) => env.ui_for_reflect_readonly_with_options(
+                            value.as_partial_reflect(),
+                            ui,
+                            id,
+                            options,
+                        ),
+                        Ok(mut value) => {
+                            ui.label(
+                                egui::RichText::new("Editing Immutable Component")
+                                    .color(egui::Color32::ORANGE),
+                            );
+                            let changed = env.ui_for_reflect_with_options(
+                                value.as_partial_reflect_mut(),
+                                ui,
+                                id,
+                                options,
+                            );
+
+                            if changed {
+                                changed_cloned_component.replace(value);
+                            }
+                        }
+                    },
+                };
+            });
 
-        #[cfg(feature = "documentation")]
-        crate::egui_utils::show_docs(_response.header_response, type_docs);
-        ui.reset_style();
+            #[cfg(feature = "documentation")]
+            crate::egui_utils::show_docs(_response.header_response, type_docs);
+            ui.reset_style();
+        };
+
+        if let Some(changed_cloned_component) = changed_cloned_component {
+            let Some(reflect_component) =
+                type_registry.get_type_data::<ReflectComponent>(component_type_id)
+            else {
+                display_err(ui, Error::NoTypeData(component_type_id, "ReflectComponent"));
+                continue;
+            };
+            let partial_reflect_component = changed_cloned_component.as_partial_reflect();
+            // TODO: SAFETY?
+            let world_mut = unsafe { world.world().world_mut() };
```

Comment:
I don't know if this is actually safe or not, or comment to put. We could remove the whole `changed_cloned_component` and move this block into the `if changed { ... }` but I was trying to only touch world outside of the extra borrow we create via 

```rust
let (mut component_view, world) = world.split_off_component((entity, component_type_id));
```